config/v1/tests/authentications.config.openshift.io/ExternalOIDC.yaml

```diff
@@ -121,6 +121,367 @@ tests:
               username:
                 claim: "preferred_username"
                 prefixPolicy: NoPrefix
+    - name: Can set OIDC providers with uid claim mapping as a specific claim
+      initial: |
+        apiVersion: config.openshift.io/v1
+        kind: Authentication
+        spec:
+          type: OIDC
+          oidcProviders:
+          - name: myoidc
+            issuer:
+              issuerURL: https://meh.tld
+              audiences: ['openshift-aud']
+            claimMappings:
+              uid:
+                claim: sub
+      expected: |
+        apiVersion: config.openshift.io/v1
+        kind: Authentication
+        spec:
+          type: OIDC
+          oidcProviders:
+          - name: myoidc
+            issuer:
+              issuerURL: https://meh.tld
+              audiences: ['openshift-aud']
+            claimMappings:
+              uid:
+                claim: sub
+    - name: Can set OIDC providers with uid claim mapping as a CEL expression
+      initial: |
+        apiVersion: config.openshift.io/v1
+        kind: Authentication
+        spec:
+          type: OIDC
+          oidcProviders:
+          - name: myoidc
+            issuer:
+              issuerURL: https://meh.tld
+              audiences: ['openshift-aud']
+            claimMappings:
+              uid:
+                expression: claims.sub
+      expected: |
+        apiVersion: config.openshift.io/v1
+        kind: Authentication
+        spec:
+          type: OIDC
+          oidcProviders:
+          - name: myoidc
+            issuer:
+              issuerURL: https://meh.tld
+              audiences: ['openshift-aud']
+            claimMappings:
+              uid:
+                expression: claims.sub
+    - name: Cannot set OIDC providers with uid claim mapping specifying both claim and expression
+      initial: |
+        apiVersion: config.openshift.io/v1
+        kind: Authentication
+        spec:
+          type: OIDC
+          oidcProviders:
+          - name: myoidc
+            issuer:
+              issuerURL: https://meh.tld
+              audiences: ['openshift-aud']
+            claimMappings:
+              uid:
+                claim: sub
+                expression: claims.sub
+      expectedError: "both claim and expression must not be set"
+    - name: Can set OIDC providers with uid claim mapping specifying an empty claim
```

Comment:
I wasn't expecting this to be allowed, why?

Reply:
I think for now I'd say lets make this API the API we want, and bring the other fields up to scratch in a separate PR, with a separate type until we can add the validation

Note as well that since the API is only graduated in HyperShift, we have much more likelihood of being able to determine what the current usage is, vs if this were enabled in the wider field.

We could also create a gate for the validations that we enable on standalone but do not enable on HyperShift, with a view to enabling the gate in the future. None of this API has shipped for standalone right?

Reply:
Correct, none of this API has shipped for standalone. The separate gate for the validations is something I hadn't considered, but is a good call out.